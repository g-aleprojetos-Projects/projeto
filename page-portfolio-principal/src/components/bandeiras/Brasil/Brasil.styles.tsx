import styled from 'styled-components';
import sizes from 'resourses/sizes';
import colors from 'resourses/colors';

export interface PropsBrasil {
  idiomaSelecionado: boolean;
}

export const Container = styled.div<PropsBrasil>`
  display: flex;
  width: ${sizes.px55};
  height: 100%;
  margin: ${sizes.px1} ${sizes.px3};
  cursor: ${props => (props.idiomaSelecionado ? 'default' : 'pointer')};
`;

export const PathVerde = styled.path<PropsBrasil>`
  fill: ${props =>
    props.idiomaSelecionado ? `${colors.verde}` : `${colors.diamont}`};
`;

export const PathAmarelo = styled.path<PropsBrasil>`
  fill: ${props =>
    props.idiomaSelecionado ? `${colors.amarelo}` : `${colors.cinza}`};
`;

export const PathAzul = styled.path<PropsBrasil>`
  fill: ${props =>
    props.idiomaSelecionado ? `${colors.pantone}` : `${colors.cinzento}`};
`;

export const PathBranco = styled.path.attrs({
  fill: `${colors.white}`,
})``;
