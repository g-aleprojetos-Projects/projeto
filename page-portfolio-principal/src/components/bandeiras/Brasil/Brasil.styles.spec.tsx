import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import * as S from './Brasil.styles';

describe('Brasil.styles', () => {
  test('Container DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a false', () => {
    const container = renderer
      .create(<S.Container idiomaSelecionado={false} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  test('Container DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a true', () => {
    const container = renderer
      .create(<S.Container idiomaSelecionado={true} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  test('PathVerde DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a false', () => {
    const container = renderer
      .create(<S.PathVerde idiomaSelecionado={false} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  test('PathVerde DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a true', () => {
    const container = renderer
      .create(<S.PathVerde idiomaSelecionado={true} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  test('PathAmarelo DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a false', () => {
    const container = renderer
      .create(<S.PathAmarelo idiomaSelecionado={false} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  test('PathAmarelo DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a true', () => {
    const container = renderer
      .create(<S.PathAmarelo idiomaSelecionado={true} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  test('PathAzul DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a false', () => {
    const container = renderer
      .create(<S.PathAzul idiomaSelecionado={false} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  test('PathAzul DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a true', () => {
    const container = renderer
      .create(<S.PathAzul idiomaSelecionado={true} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  test('PathBranco DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.PathBranco />).toJSON();
    expect(container).toMatchSnapshot();
  });
});
