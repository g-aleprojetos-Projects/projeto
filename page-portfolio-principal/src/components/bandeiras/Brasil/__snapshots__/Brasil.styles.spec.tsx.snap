// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brasil.styles Container DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a false 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 3.438rem;
  height: 100%;
  margin: 0.063rem 0.188rem;
  cursor: pointer;
}

<div
  className="c0"
/>
`;

exports[`Brasil.styles Container DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a true 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 3.438rem;
  height: 100%;
  margin: 0.063rem 0.188rem;
  cursor: default;
}

<div
  className="c0"
/>
`;

exports[`Brasil.styles PathAmarelo DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a false 1`] = `
.c0 {
  fill: #C8C8C8;
}

<path
  className="c0"
/>
`;

exports[`Brasil.styles PathAmarelo DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a true 1`] = `
.c0 {
  fill: #FFE15A;
}

<path
  className="c0"
/>
`;

exports[`Brasil.styles PathAzul DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a false 1`] = `
.c0 {
  fill: #3C3C3C;
}

<path
  className="c0"
/>
`;

exports[`Brasil.styles PathAzul DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a true 1`] = `
.c0 {
  fill: #41479B;
}

<path
  className="c0"
/>
`;

exports[`Brasil.styles PathBranco DEVE ser igual ao snapshot 1`] = `
<path
  className=""
  fill="#ffffff"
/>
`;

exports[`Brasil.styles PathVerde DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a false 1`] = `
.c0 {
  fill: #9B9C9A;
}

<path
  className="c0"
/>
`;

exports[`Brasil.styles PathVerde DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a true 1`] = `
.c0 {
  fill: #73AF00;
}

<path
  className="c0"
/>
`;
